src/core/execute.c

```diff
@@ -2400,12 +2400,17 @@ static int apply_mount_namespace(
          * that with a special, recognizable error ENOANO. In this case, silently proceeed, but only if exclusively
          * sandboxing options were used, i.e. nothing such as RootDirectory= or BindMount= that would result in a
          * completely different execution environment. */
-        if (r == -ENOANO &&
-            n_bind_mounts == 0 && context->n_temporary_filesystems == 0 &&
-            !root_dir && !root_image &&
-            !context->dynamic_user) {
-                log_unit_debug(u, "Failed to set up namespace, assuming containerized execution and ignoring.");
-                return 0;
+        if (r == -ENOANO) {
+                if (n_bind_mounts == 0 &&
+                    context->n_temporary_filesystems == 0 &&
+                    !root_dir && !root_image &&
+                    !context->dynamic_user) {
+                        log_unit_debug_errno(u, errno, "Failed to set up namespace, assuming containerized execution and ignoring: %m");
+                        return 0;
+                }
+
+                /* Use original errno */
+                return -errno;
```

Comment:
urks, i can't say i like this. we generally don't use errno to pass errors, and this way we make the fact that EANO means we called unshare() right before and it failed and set errno API of setup_namespace(). But to me that's too much internal behaviour of setup_namespace() becoming API... errno handling should be hidden in the function i think and not be relied upon from outsde...

can't we find a nicer way for this?

I mean, i see where you are coming from, but this at least needs an comment on the sending side (i.e. in setup_namespace where we return EANO) and on the receiveing side, about this special error passing

Reply:
I cannot find a nicer way... Of course, adding a pointer argument e.g., `bool *failure_by_unshare` or something can avoid using errno. But it seems ugly.

I will add comments about that on both sender and receiver.